### PR TITLE
Kustomise support 

### PR DESCRIPTION
**Pull request by metahertz**
Kustomize support with re-factored K8s runner (after the Kubernetes graph check integration).
- Expecting to need another rebase, so expecting some tests to fail on PR initially 
- Comments/feedback welcome

K8S runner now has two "Mutate" functions, where report generation logic has moved. (One for CKV_ and one for CKV2_ report items). These can be overwritten by other Kubernetes-related frameworks (helm, kustomize) via Class inheritance, in order to modify the report data after passing discovered k8s resource to the k8s runner.

This is because in the case of dependant frameworks, the temporary files generated by a templating operation are not what the user cares about, instead the file and path directives need to point towards a different item (such as the initial template, or the Kustomization.yaml item).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by metahertz** [comment]
Rebased, linter issues resolved and tests passing. Open to Q&A / Review / Feedback.

**Comment by metahertz** [comment]
Will add documentation and usage docs for checkov.io to the PR tomorrow!

**Comment by schosterbarak** [comment]
you'll need to add kustomize to: https://github.com/bridgecrewio/checkov/blob/master/Dockerfile and https://github.com/bridgecrewio/checkov/blob/master/.github/workflows/pr-test.yml#L70

File: .github/workflows/pr-test.yml [hunk 1]
```diff
@@ -48,6 +48,7 @@ jobs:
         run: |
           pipenv --python ${{ matrix.python }}
           pipenv install --dev
+      - uses: imranismail/setup-kustomize@v1
```
Comment (schosterbarak) [inline .github/workflows/pr-test.yml:51]:
```suggestion
      - uses: imranismail/setup-kustomize@v1
          if: ${{ runner.os != 'windows' }}
```

**Comment by exocode** [comment]
This integration will be awesome. We all use Kustomize.
Just wanna drop some motivation 🚀 🚀 and magic dust

**Comment by exocode** [comment]
this is merged now! Yay! ... but I do not know how to use it . I found that docs https://github.com/bridgecrewio/checkov/blob/master/docs/7.Scan%20Examples/Kustomize.md

and applied that to my folders containing the regular /kustomize structure: 
```
/backend
  /kustomize
    /base
      /kustomization.yaml
      /deployment.yaml
    /overlays
      /staging
        /kustomization.yaml
        /deployment.yaml
      /production
        /kustomization.yaml
        /deployment.yaml
```
and running the example command
```
checkov -d ./backend/ --framework kustomize --check CKV_K8S_43 --no-guide --quiet --compact
More details: https://www.bridgecrew.cloud/projects?repository=jezekjan_cli_repo/backend&branch=bc-592b146_master&runId=latest
```

I also tried to sequentially reduced the command from the back, like so: 

```
❯ checkov -d ./app/backend/ --framework kustomize --compact

       _               _
   ___| |__   ___  ___| | _______   __
  / __| '_ \ / _ \/ __| |/ / _ \ \ / /
 | (__| | | |  __/ (__|   < (_) \ V /
  \___|_| |_|\___|\___|_|\_\___/ \_/

By bridgecrew.io | version: 2.0.756

More details: https://www.bridgecrew.cloud/projects?repository=jezekjan_cli_repo/backend&branch=bc-44d95d9_master&runId=latest
```

I also tried different folders: 

```
❯ checkov -d ./app/backend/kustomize --framework kustomize
❯ checkov -d ./app/backend/ --framework kustomize
❯ checkov -d ./app/ --framework kustomize
❯ checkov -d ./ --framework kustomize

```
All commands returns only the link to the webUI

I don't see any output in the CLI. Is this intentionally? I thought there is an output, like in the docs. 

Maybe my version does not match or I need another kustomize binary or similar. 

I installed `kustomize` via Homebrew: 
```
❯ brew install kustomize
❯ kustomize version
{Version:kustomize/v4.4.1 GitCommit:b2d65ddc98e09187a8e38adc27c30bab078c1dbf BuildDate:2021-11-11T23:27:14Z GoOs:darwin GoArch:amd64}
```

and installed checkov via `pip3`

```
pip3 install -U checkov
❯ checkov --version
2.0.756
```

I know the feature brand new, but only wanna be sure that I am not using it right. 

If I run checkov against single files it works as expected: 

```
❯ checkov -f ./app/backend/kustomize/overlays/staging/backend-deployment.yml

       _               _
   ___| |__   ___  ___| | _______   __
  / __| '_ \ / _ \/ __| |/ / _ \ \ / /
 | (__| | | |  __/ (__|   < (_) \ V /
  \___|_| |_|\___|\___|_|\_\___/ \_/

By bridgecrew.io | version: 2.0.756

kubernetes scan results:

Passed checks: 74, Failed checks: 15, Skipped checks: 0

Check: CKV_K8S_80: "Ensure that the admission control plugin AlwaysPullImages is set"
	PASSED for resource: Deployment.default.backend
	File: /app/backend/kustomize/overlays/staging/backend-deployment.yml:1-23
	Guide: https://docs.bridgecrew.io/docs/ensure-that-the-admission-control-plugin-alwayspullimages-is-set

Check: CKV_K8S_113: "Ensure that the --bind-address argument is set to 127.0.0.1"
	PASSED for resource: Deployment.default.backend
	File: /app/backend/kustomize/overlays/staging/backend-deployment.yml:1-23
	Guide: https://docs.bridgecrew.io/docs/ensure-that-the-bind-address-argument-is-set-to-127001

```

PS: I run all commands from the root of my project folder.

Any help would be awesome. Thank you

**Pull request merged**